libs/ml/include/ml/core/node.hpp

```diff
@@ -52,6 +52,8 @@ class Node
   using VecTensorType    = typename fetch::ml::ops::Ops<TensorType>::VecTensorType;
   using SPType           = fetch::ml::NodeSaveableParams<TensorType>;
   using NodeErrorMapType = std::unordered_map<Node<TensorType> *, std::vector<TensorType>>;
+  using ShapeType =
```

Comment:
ShapeType is ALWAYS a math::SizeVector everywhere, and you should never explicitly set uint64_t for a SizeType or ShapeType since we should be allowed to change this to a uint32 or uint128 and everything should still work fine. Therefore just use SizeVector

Reply:
Sure! I'll update type system for all use cases.

Reply:
done